parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -203,12 +158,12 @@ pub mod pallet {
 		}
 	}
 
-	impl<T: Config> Pallet<T>
-	where
-		T::AccountId: AsRef<[u8]>,
-	{
+	impl<T: Config> Pallet<T> {
 		/// Submit message on the outbound channel
-		pub fn submit(who: &T::AccountId, target: H160, payload: &[u8]) -> DispatchResult {
+		// TODO: Remove target once incentivized channel & app removal are done
+		pub fn submit(who: &T::SourceId, _target: H160, payload: &[u8]) -> DispatchResult {
+			// TODO: Do we need to convert the error values passed to ensure! to
+			// sp_runtime::DispatchError? Does the ensure! macro hide a possible type error here?
```

Comment:
Pallet errors such as `Error::<T>::QueueSizeLimitReached` implement the necessary traits to convert to DispatchError. So not needed.

Reply:
The `ensure!` macro would not compile if the passed error was not convertible to DispatchError